[jit] allow bools to be used as attributes

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#19440 [jit] allow bools to be used as attributes**

As title. Required adding NEWTRUE and NEWFALSE opcode handling to the
pickler. Fixes #19441

Differential Revision: [D15005723](https://our.internmc.facebook.com/intern/diff/D15005723)

suo committed:
[jit] allow bools to be used as attributes

As title. Required adding NEWTRUE and NEWFALSE opcode handling to the
pickler

driazati commented on torch/csrc/jit/pickler.cpp:
```diff
@@ -449,7 +455,7 @@ OpCode Unpickler::readInstruction() {
     } break;
     default:
-      AT_ERROR("Unknown opcode for unpickling");
+      AT_ERROR("Unknown opcode for unpickling: ", static_cast<char>(opcode));
```
The opcodes aren't always printable characters, maybe a `uint8_t` better

suo committed:
comment on "[jit] allow bools to be used as attributes"

[jit] allow bools to be used as attributes

As title. Required adding NEWTRUE and NEWFALSE opcode handling to the
pickler

gh-metadata: pytorch pytorch 19440 gh/suo/28/head
